File: src/main/java/com/back/domain/post/post/entity/Post.java
package com.back.domain.post.post.entity;

import com.back.domain.category.category.entity.Category; // 카테고리 임포트 확인
import com.back.domain.member.member.entity.Member;
import com.back.global.jpa.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Post extends BaseEntity {
    @ManyToOne(fetch = FetchType.LAZY)
    private Member seller;

    private String title;
    @Column(columnDefinition = "TEXT")
    private String content;
    private int price;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private PostStatus status = PostStatus.SALE;

    @Builder.Default
    private boolean deleted = false;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    @OneToMany(mappedBy = "post", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<PostImage> postImages = new ArrayList<>();

    @Column(columnDefinition = "BIGINT DEFAULT 0")
    @Builder.Default
    private long viewCount = 0L;

    public void addPostImage(PostImage postImage) {
        this.postImages.add(postImage);
    }


    public void update(String title, String content, int price, Category category) {
        this.title = title;
        this.content = content;
        this.price = price;
        this.category = category;
    }

    public void increaseViewCount() {
        this.viewCount++;
    }

    public void updateStatus(PostStatus status) {
        this.status = status;
    }
}

File: src/main/java/com/back/domain/post/post/entity/Post.kt
package com.back.domain.post.post.entity

import com.back.domain.category.category.entity.Category
import com.back.domain.member.member.entity.Member
import com.back.global.jpa.entity.BaseEntity
import jakarta.persistence.*

@Entity
class Post(
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "seller_id")
    val seller: Member,

    @Column(length = 50, nullable = false)
    var title: String,

    @Column(columnDefinition = "TEXT", nullable = false)
    var content: String,

    var price: Int,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    var category: Category,

    @Enumerated(EnumType.STRING)
    var status: PostStatus = PostStatus.SALE,

    var deleted: Boolean = false

) : BaseEntity() {

    @OneToMany(mappedBy = "post", cascade = [CascadeType.ALL], orphanRemoval = true)
    var postImages: MutableList<PostImage> = mutableListOf()
        protected set // 외부에서 리스트 자체를 통째로 교체하지 못하게 보호

    @Column(columnDefinition = "BIGINT DEFAULT 0")
    var viewCount: Long = 0L
        protected set

    fun addPostImage(postImage: PostImage) {
        this.postImages.add(postImage)
    }

    fun update(title: String, content: String, price: Int, category: Category) {
        this.title = title
        this.content = content
        this.price = price
        this.category = category
    }

    fun increaseViewCount() {
        this.viewCount++
    }

    fun updateStatus(status: PostStatus) {
        this.status = status
    }
}